django_tasks/task.py

```diff
@@ -226,21 +227,34 @@ class TaskResult(Generic[T]):
     backend: str
     """The name of the backend the task will run on"""
 
-    _result: Optional[Union[T, dict]] = field(init=False, default=None)
+    _result: Optional[Union[T, SerializedExceptionDict]] = field(
+        init=False, default=None
+    )
 
     @property
     def result(self) -> Optional[Union[T, BaseException]]:
         if self.status == ResultStatus.COMPLETE:
             return cast(T, self._result)
         elif self.status == ResultStatus.FAILED:
             return (
-                exception_from_dict(cast(dict, self._result))
+                exception_from_dict(cast(SerializedExceptionDict, self._result))
                 if self._result is not None
                 else None
             )
 
         raise ValueError("Task has not finished yet")
 
+    @property
+    def traceback(self) -> Optional[Union[T, List[str]]]:
+        if self.status == ResultStatus.COMPLETE:
+            return None
+        if self.status == ResultStatus.FAILED:
+            if self._result is None:
+                return None
+            return cast(SerializedExceptionDict, self._result)["exc_traceback"]
```

Comment:
Question: Are there cases where there wouldn't be a traceback?

Reply:
If we didn't think of an edge case when producing the traceback, this means a bug on our side would not crash task retrieval, only provide no information.

I don't know if it's worth keeping. You choose.

Reply:
There is one thing to consider. In `test_complex_exception()`, the task is marked as failed yet `._result` is `None`. In that case, the task is marked as `FAILED`, but no dict is stored in `._result`:

```python
  def test_complex_exception(self) -> None:
        ....
        self.assertEqual(result.status, ResultStatus.FAILED)

        self.assertIsNone(result.result)
        self.assertIsNone(result.traceback) # this would fail if it assumes a dict is always there
        ...
  ```

So either I keep the check that the result is not `None`, or the behavior for complex exceptions must change.

Reply:
For now I kept the original behavior, let me know which option you prefer.

Reply:
Yeah if we can't serialize the exception, `None` is fine, which means the code here is fine. I think you're right there's not likely to be a case where there's an exception and no traceback.